File: src/main/java/Ex05_Nullability.java
import java.util.Optional;

public class Ex05_Nullability {
    public static void main(String[] args) {
        String rawValue = nullHalfTheTime();
        String filteredValue = rawValue != null ? rawValue : "";
        if (!filteredValue.equals("")) System.out.println("First " + filteredValue);

        Optional.ofNullable(nullHalfTheTime())
                .map(String::toLowerCase)
                .ifPresent(it -> System.out.println("Second " + it));
    }

    static String nullHalfTheTime() {
        long rand = Math.round(Math.random()); // Provide us 0 or 1 number
        return rand == 0 ? "I love Java" : null;
    }
}

File: src/main/kotlin/Ex05_Nullability.kt
import kotlin.math.roundToInt

fun main() {
  val result: String? = nullHalfTheTime()
  println(result?.lowercase())
  // Or we can pass a default value
  val default: String = nullHalfTheTime() ?: "I love Java"
  println(default)
}

fun nullHalfTheTime(): String? {
  val rand: Int = Math.random().roundToInt() // Provide us 0 or 1 number
  return if (rand == 0) "I love Kotlin" else null
}